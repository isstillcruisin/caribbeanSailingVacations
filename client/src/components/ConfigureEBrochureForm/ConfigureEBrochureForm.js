import React, {Component} from "react";
import { Table, Row, Col, Button } from 'react-bootstrap';
import BootstrapSwitchButton from 'bootstrap-switch-button-react'
import { LinkContainer } from 'react-router-bootstrap'

class ConfigureEBrochureForm extends Component {
  render() {
    return <div style={{color: 'black'}}>
      <form>
        <div className="form-group edit-e-brochure-form">
          <label htmlFor="e-brochure-name">
            <strong>EBrochure Name</strong>
          </label>
          <input
            className="form-control"
            id="e-brochure-name"
            type="text"
            value={this.props.eBrochure.name}
            name="eBrochureName"
            required
          />
        </div>
      </form>

      <Table striped bordered hover>
        <thead>
          <tr>
            <th>Name</th>
            <th>Enabled?</th>
            <th>Yacht Details</th>
          </tr>
        </thead>
        <tbody>
          {this.renderAllYachtRows(this.props.allYachts, this.props.eBrochure)}
        </tbody>
      </Table>
    </div>
  }

  eBrochureIncludesYacht = (eBrochure, yacht) => {
    return eBrochure.yachts.map(y => y._id).includes(yacht._id);
  }

  renderAllYachtRows = (allYachts, eBrochure) => {
    return allYachts.map((yacht, i) => {
      return <tr key={`yacht-${i}`}>
          <td>{yacht.boatName}</td>
          <td>
            <BootstrapSwitchButton
              checked={this.eBrochureIncludesYacht(eBrochure, yacht)}
              onlabel='Y'
              offlabel='N'
              onChange={(checked: boolean) => {
                if (checked) {
                  this.props.handleEnableYacht(yacht)
                } else {
                  this.props.handleDisableYacht(yacht) 
                }
              }}
            />
          </td>
          <td>
            <LinkContainer
              params={{ id: yacht._id }}
              to={`/boat/${yacht._id}`}
              className="boat-detail-link"
            >
              <Button>See Details</Button>
            </LinkContainer>
          </td> 
        </tr>
    });
  }
}

export default ConfigureEBrochureForm;
