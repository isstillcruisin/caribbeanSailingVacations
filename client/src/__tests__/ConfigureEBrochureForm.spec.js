import React from "react";
import { create } from "react-test-renderer";
import ConfigureEBrochureForm from '../components/ConfigureEBrochureForm';
import { MemoryRouter } from 'react-router';

describe("ConfigureEBrochureForm component", () => {
  test("Matches the snapshot", () => {
    const form = create(
       <MemoryRouter>
        <ConfigureEBrochureForm 
          allYachts={
            [
              {'_id': 1, boatName: 'boat1', maxPassengers: 5, pricePerWeek: 1000, imgs: []}, 
              {'_id': 2, boatName: 'boat2', maxPassengers: 5, pricePerWeek: 1000, imgs: []}, 
            ]
          }
          eBrochure={
            {
              yachts:[
                {_id: 2, boatName: 'boat2', maxPassengers: 5, pricePerWeek: 1000, imgs: []}, 
              ],
              name: 'test'
            }
          }
        />
      </MemoryRouter>
    );
    expect(form.toJSON()).toMatchSnapshot();
  });
});