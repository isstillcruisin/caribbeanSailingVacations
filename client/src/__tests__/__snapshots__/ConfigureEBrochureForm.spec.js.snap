// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigureEBrochureForm component Matches the snapshot 1`] = `
<div
  style={
    Object {
      "color": "black",
    }
  }
>
  <form>
    <div
      className="form-group edit-e-brochure-form"
    >
      <label
        htmlFor="e-brochure-name"
      >
        <strong>
          EBrochure Name
        </strong>
      </label>
      <input
        className="form-control"
        disabled={true}
        id="e-brochure-name"
        name="eBrochureName"
        type="text"
        value="test"
      />
    </div>
  </form>
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Maximum Number Of Passengers
        </th>
        <th>
          Price Per Week
        </th>
        <th>
          Images
        </th>
        <th>
          Enabled?
        </th>
        <th>
          More Yacht Details
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          boat1
        </td>
        <td>
          5
        </td>
        <td>
          $1,000.00
        </td>
        <td>
          <div
            className="image-thumbnails-in-table"
          />
        </td>
        <td>
          <div
            className="switch btn off btn-light"
            onClick={[Function]}
            style={Object {}}
          >
            <div
              className="switch-group"
            >
              <label
                className="switch-on btn btn-primary"
                style={Object {}}
              >
                Y
              </label>
              <label
                className="switch-off btn btn-light"
                style={Object {}}
              >
                N
              </label>
              <span
                className="switch-handle btn btn-light"
              />
            </div>
          </div>
        </td>
        <td>
          <a
            className="boat-detail-link btn btn-primary"
            href="/boat/1"
            onClick={[Function]}
            onKeyDown={[Function]}
            params={
              Object {
                "id": 1,
              }
            }
          >
            See Details
          </a>
        </td>
      </tr>
      <tr>
        <td>
          boat2
        </td>
        <td>
          5
        </td>
        <td>
          $1,000.00
        </td>
        <td>
          <div
            className="image-thumbnails-in-table"
          />
        </td>
        <td>
          <div
            className="switch btn on btn-primary"
            onClick={[Function]}
            style={Object {}}
          >
            <div
              className="switch-group"
            >
              <label
                className="switch-on btn btn-primary"
                style={Object {}}
              >
                Y
              </label>
              <label
                className="switch-off btn btn-light"
                style={Object {}}
              >
                N
              </label>
              <span
                className="switch-handle btn btn-light"
              />
            </div>
          </div>
        </td>
        <td>
          <a
            className="boat-detail-link btn btn-primary"
            href="/boat/2"
            onClick={[Function]}
            onKeyDown={[Function]}
            params={
              Object {
                "id": 2,
              }
            }
          >
            See Details
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
